### PR TITLE
Manage books and notes (see Issue #82)

### PR DESCRIPTION
**Pull request by SarahKurek**
**Description**
Implemented deleting books and notes. When element is selected by long click, it can be deleted using the button in the menu.

Implemented changing the book data of an existing book. Therefore the existing BookFormFragment was adapted.
When a book is selected by long click, it's data can be changed using the button in the menu.

**Affects**
BookFormFragment, BookNotesView, BookFragment and the corresponding Models and DAOs.

**Notes for Reviewer**
Please review & comment/approve.

### CODE REVIEW

**Comment by claudia3** [comment]
When I delete a book and try to open the shelf where the book was deleted again the app crashes.

**Comment by claudia3** [comment]
The app crashes when you click "ändern" when you have an empty selection in BookFragment.

_E/AndroidRuntime: FATAL EXCEPTION: main
    Process: de.bibbuddy, PID: 9181
    java.lang.IndexOutOfBoundsException: Index: 0, Size: 0
        at java.util.ArrayList.get(ArrayList.java:437)
        at de.bibbuddy.BookFragment.handleChangeBookData(BookFragment.java:122)
        at de.bibbuddy.BookFragment.onOptionsItemSelected(BookFragment.java:94)
        at androidx.fragment.app.Fragment.performOptionsItemSelected(Fragment.java:3109)
        at androidx.fragment.app.FragmentManager.dispatchOptionsItemSelected(FragmentManager.java:3210)
        at androidx.fragment.app.FragmentController.dispatchOptionsItemSelected(FragmentController.java:416)
        at androidx.fragment.app.FragmentActivity.onMenuItemSelected(FragmentActivity.java:346)
        at androidx.appcompat.app.AppCompatActivity.onMenuItemSelected(AppCompatActivity.java:228)
        at androidx.appcompat.view.WindowCallbackWrapper.onMenuItemSelected(WindowCallbackWrapper.java:109)
        at androidx.appcompat.view.WindowCallbackWrapper.onMenuItemSelected(WindowCallbackWrapper.java:109)
        at androidx.appcompat.app.ToolbarActionBar$2.onMenuItemClick(ToolbarActionBar.java:65)
        at androidx.appcompat.widget.Toolbar$1.onMenuItemClick(Toolbar.java:207)
        at androidx.appcompat.widget.ActionMenuView$MenuBuilderCallback.onMenuItemSelected(ActionMenuView.java:779)
        at androidx.appcompat.view.menu.MenuBuilder.dispatchMenuItemSelected(MenuBuilder.java:834)
        at androidx.appcompat.view.menu.MenuItemImpl.invoke(MenuItemImpl.java:158)
        at androidx.appcompat.view.menu.MenuBuilder.performItemAction(MenuBuilder.java:985)
        at androidx.appcompat.view.menu.MenuPopup.onItemClick(MenuPopup.java:128)
        at android.widget.AdapterView.performItemClick(AdapterView.java:330)
        at android.widget.AbsListView.performItemClick(AbsListView.java:1187)
        at android.widget.AbsListView$PerformClick.run(AbsListView.java:3179)
        at android.widget.AbsListView$3.run(AbsListView.java:4097)
        at android.os.Handler.handleCallback(Handler.java:938)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:223)
        at android.app.ActivityThread.main(ActivityThread.java:7656)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
I/Process: Sending signal. PID: 9181 SIG: 9_

**Comment by claudia3** [comment]
Please adjust the menu in the BookFragment depending on the user selection.

File: BibBuddy/app/src/main/java/de/bibbuddy/LibraryFragment.java [hunk 1]
```diff
@@ -364,6 +366,17 @@ public void onItemClicked(int position) {
     updateBookListView(libraryItem);
   }
 
+  @Override
+  public void onLongItemClicked(int position, BookItem bookItem, View v) {
```
Comment (claudia3) [inline BibBuddy/app/src/main/java/de/bibbuddy/LibraryFragment.java:370]: Why is this function also in the LibraryFragment?

**Review by claudia3 — CHANGES_REQUESTED** [review]
Please fix the commented problems before I take a look at the code.

File: BibBuddy/app/src/main/java/de/bibbuddy/BookFormFragment.java [hunk 2]
```diff
@@ -40,21 +40,78 @@ public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup c
...
-    setupButtons(view);
     setupInput(view);
 
     Bundle bundle = this.getArguments();
-    shelfName = bundle.getString(LibraryKeys.SHELF_NAME);
-    shelfId = bundle.getLong(LibraryKeys.SHELF_ID);
+    if (bundle != null) {
+      shelfName = bundle.getString(LibraryKeys.SHELF_NAME);
+      shelfId = bundle.getLong(LibraryKeys.SHELF_ID);
+      long bookId = bundle.getLong(LibraryKeys.BOOK_ID, 0);
```
Comment (claudia3) [inline BibBuddy/app/src/main/java/de/bibbuddy/BookFormFragment.java:49]:
```suggestion
      Long bookId = bundle.getLong(LibraryKeys.BOOK_ID, 0);
```

Comment (SarahKurek) [reply BibBuddy/app/src/main/java/de/bibbuddy/BookFormFragment.java:49]: belongs like this

File: BibBuddy/app/src/main/java/de/bibbuddy/LibraryFragment.java [hunk 3]
```diff
@@ -63,6 +64,7 @@ public void handleOnBackPressed() {
 
     setHasOptionsMenu(true);
     selectedShelfItems = new ArrayList<ShelfItem>();
+    selectedBookItems = new ArrayList<BookItem>();
```
Comment (claudia3) [inline BibBuddy/app/src/main/java/de/bibbuddy/LibraryFragment.java:67]: Why do you create a selectedBookItems List in the LibraryFragment? In the library you can't select a book. You can only select shelves.

File: BibBuddy/app/src/main/java/de/bibbuddy/LibraryFragment.java [hunk 4]
```diff
@@ -38,6 +38,7 @@
   private LibraryModel libraryModel;
   private LibraryRecyclerViewAdapter adapter;
   private List<ShelfItem> selectedShelfItems;
+  private List<BookItem> selectedBookItems;
```
Comment (claudia3) [inline BibBuddy/app/src/main/java/de/bibbuddy/LibraryFragment.java:41]: Why do you create a selectedBookItems List in the LibraryFragment? In the library you can't select a book. You can only select shelves.

**Review by claudia3 — CHANGES_REQUESTED** [review]
Please remove the unnecessary selectedBookItems List in the Library and the corressponding functions.

**Review by claudia3 — APPROVED** [review]
Thanks for the changes. The added features work.

**Pull request merged**